### PR TITLE
REST shorter prettier

### PR DESCRIPTION
**Pull request by michalwski**
This PR focuses only on the REST API for backend services. The main change is the URL generator which now creates shorter and nicer URLs.

### CODE REVIEW

File: apps/ejabberd/src/mongoose_admin.erl [hunk 1]
```diff
@@ -181,9 +166,8 @@ register(User, Host, Password) ->
             throw({error, String})
     end.
 
-unregister(Jid) ->
-    J = jid:from_binary(Jid),
-    ejabberd_auth:remove_user(J#jid.luser, J#jid.lserver),
+unregister(Host, User) ->
+    ejabberd_auth:remove_user(User, Host),
     <<"">>.
```
Comment (Dzol) [inline apps/ejabberd/src/mongoose_admin.erl]: Perhaps we could return something more meaningful than `<<"">>` (even it doesn't get into an HTTP response).

Comment (michalwski) [inline apps/ejabberd/src/mongoose_admin.erl]: Yes, I think simple `ok` would be enough here.

File: apps/ejabberd/src/mongoose_commands.erl [hunk 2]
```diff
@@ -108,14 +108,15 @@
...
           module :: module(),                         %% module to call
           function :: atom(),                         %% function to call
-          action :: action(),                 %% so that the HTTP side can decide which verb to require
+          action :: action(),                         %% so that the HTTP side can decide which verb to require
           args = [] :: [argspec()],                   %% this is both for introspection and type check on call
           caller_pos :: integer(),                    %% internal use
-          identifiers = [] :: [atom()],               %% if action is 'update' then it must be a subset of args
+          identifiers = [] :: [atom()],               %% resource identifiers, a subset of args
           security_policy = [admin] :: security(),    %% permissions required to run this command
           result :: argspec()|ok                      %% what the called func should return; if ok then return of called
```
Comment (Dzol) [inline apps/ejabberd/src/mongoose_commands.erl]: We could change returning the atom `ok` to returning `ignore`.

Comment (michalwski) [inline apps/ejabberd/src/mongoose_commands.erl]: Why would it be better, could you explain?

Comment (Dzol) [inline apps/ejabberd/src/mongoose_commands.erl]: A spec like `{..., {result, ignore}]}` communicates that the return value is actually ignored whereas currently that's not so clear with `{..., {result, ok}]}`.

Comment (michalwski) [inline apps/ejabberd/src/mongoose_commands.erl]: In fact I personally think that all the methods should always return `ok` or `{ok, STH_MORE}`. And the decisions whether or not ignore this should be left to the caller. But that's I think a candidate for separate PR.

Comment (Dzol) [inline apps/ejabberd/src/mongoose_commands.erl]: That sounds better to me too.

**Pull request merged**